docs/using-hive.md

```diff
@@ -2,162 +2,326 @@
-### Using Global Pull Secret
-
-GlobalPullSecret can be used to specify a pull secret that will be used globally by all of the cluster deployments created by Hive.
-When GlobalPullSecret is defined in Hive namespace and a cluster deployment specific pull secret is specified, the registry authentication in both secrets will be merged and used by the new OpenShift cluster.
-When a registry exists in both pull secrets, precedence will be given to the contents of the cluster specific pull secret.
-
-The global pull secret must live in the "hive" namespace, to create one:
+Cluster provisioning begins when a caller creates a `ClusterDeployment` CR, which is the core Hive resource used to control the lifecycle of a cluster and the Hive API entrypoint.
+
+Hive comes with an optional `hiveutil` binary to assist creating the `ClusterDeployment` and its dependencies. See the [hiveutil documentation](hiveutil.md) for more information.
+
+### DNS
+
+OpenShift installation requires a live and functioning DNS zone in the cloud account into which you will be installing the new cluster(s). For example if you own example.com, you could create a hive.example.com subdomain in Route53, and ensure that you have made the appropriate NS entries under example.com to delegate to the Route53 zone. When creating a new cluster, the installer will make future DNS entries under hive.example.com as needed for the cluster(s).
+
+In addition to the default OpenShift DNS support, Hive offers a DNS feature called Managed DNS. With Managed DNS, Hive can automatically create delegated zones for approved base domains. For example, if hive.example.com exists and is specified as your managed domain, you can specify a base domain of cluster1.hive.example.com on your `ClusterDeployment`, and Hive will create this zone for you, add forwarding records in the base domain, wait for it to resolve, and then proceed with installation.
+
+### Pull Secret
+
+OpenShift installation requires a pull secret obtained from try.openshift.com. You can specify an individual pull secret for each cluster Hive creates, or you can use a global pull secret that will be used by all of the clusters Hive creates.
 
 ```bash
-oc create secret generic global-pull-secret --from-file=.dockerconfigjson=<path>/config.json --type=kubernetes.io/dockerconfigjson --namespace hive
+oc create secret generic mycluster-pull-secret --from-file=.dockerconfigjson=/path/to/pull-secret --type=kubernetes.io/dockerconfigjson --namespace mynamespace
 ```
 
-Edit the HiveConfig to add global pull secret.
+```yaml
+apiVersion: v1
+data:
+  .dockerconfigjson: REDACTED
+kind: Secret
+metadata:
+  name: mycluster-pull-secret
+  namespace: hive
+type: kubernetes.io/dockerconfigjson
+```
+
+When a global pull secret is defined in the `hive` namespace and a `ClusterDeployment`-specific pull secret is specified, the registry authentication in both secrets will be merged and used by the new OpenShift cluster.
```

Comment:
We've lost the header for global pull secret, which takes away a section we can link to specifically (which I just used with QE this morning). IMO this should be restored, I don't think regular pull secrets and global pull secrets should be lumped into one section, I prefer seeing global called out as a specific feature you can use if you wish.